lib/protocols/lister.ex

```diff
@@ -0,0 +1,35 @@
+defprotocol Lister do
+  def to_list(data)
+end
+
+defimpl Lister, for: List do
+  def to_list(list), do: list
+end
+
+defimpl Lister, for: Tuple do
+  defdelegate to_list(tuple), to: Tuple
+end
+
+defimpl Lister, for: Map do
+  defdelegate to_list(map), to: Enum
+end
+
+defimpl Lister, for: BitString do
+  def to_list(string), do: String.split(string, "", trim: true)
+end
+
+defimpl Lister, for: Atom do
+  def to_list(atom), do: Atom.to_string(atom) |> String.split("", trim: true)
```

Comment:
У вас же есть уже имплементация для строки, ей и отдайте.
```suggestion
  def to_list(atom), do: atom |> Atom.to_string() |> Lister.to_list()
```